src/sagemaker_sklearn_extension/feature_extraction/sequences.py

```diff
@@ -393,30 +445,76 @@ def _hybrid_interpolation(x):
+            default_fc_parameters=min_settings,
             column_id="id",
             column_sort="time",
             impute_function=impute,
+            n_jobs=0,
         )
+        self.min_settings_card = tsfresh_features.shape[1]
+        # Minimal features computed indepdently to ensure they go first in the output,
+        # this is needed to ensure their survival when filtering features
+        if self.extraction_type in ["efficient", "all"]:
+            if self.extraction_type == "efficient":
+                settings = EfficientFCParameters()
+            else:
+                settings = ComprehensiveFCParameters()
+            settings = {k: v for k, v in settings.items() if k not in min_settings}
+            tsfresh_features_extra = extract_features(
+                X_df_no_nans,
+                default_fc_parameters=settings,
+                column_id="id",
+                column_sort="time",
+                impute_function=impute,
+                n_jobs=0,
+            )
+            self.extra_settings_card = tsfresh_features_extra.shape[1]
+            tsfresh_features = pd.concat([tsfresh_features, tsfresh_features_extra], axis=1)
+
         # If X_df.dropna() dropped some observations entirely (i.e., due to all NaNs),
         # impute each tsfresh feature for those observations with the median of that tsfresh feature
         tsfresh_features_imputed = impute(tsfresh_features.reindex(pd.RangeIndex(X_df["id"].max() + 1)))
         return tsfresh_features_imputed, X_df
 
+    def _filter_features(self, tsfresh_features, mode="transform", transform_thresholds=None):
+        if self.expansion_threshold < self.min_settings_card:
+            raise ValueError(
+                f"Provided filter threshold(s) (= {self.expansion_threshold}) can not be smaller than "
+                f"number of features generated by minimal settings (= {self.min_settings_card})"
+            )
+        filter_order = np.arange(self.min_settings_card, tsfresh_features.shape[1])
+        random_state = np.random.get_state()
+        np.random.seed(self.feature_sampling_seed)
+        np.random.shuffle(filter_order)
+        np.random.set_state(random_state)
+        survivors = list(range(self.min_settings_card)) + list(
+            filter_order[: self.expansion_threshold - self.min_settings_card]
+        )
+        tsfresh_features = tsfresh_features.iloc[:, survivors]
+
+        if mode == "transform":
+            if len(transform_thresholds) != tsfresh_features.shape[0]:
+                raise ValueError(
+                    f"In 'transform' mode transform_thresholds should have number of entries "
+                    f"(= {len(transform_thresholds)}) that corresponds to the number of records "
+                    f"in tsfresh_features (= {tsfresh_features.shape[0]})."
+                )
+            for thrsh_i, thrsh in enumerate(transform_thresholds):
+                tsfresh_features.iloc[thrsh_i, thrsh:] = 0
```

Comment:
I don't get something. You filter features here base on thresholds individual to the sequences you see. That's fine. Before this, you filter features no matter what using `self. expansion_threshold`. What is the point of that? Why not just set `expansion_threshold = infinity` and get rid of the complication with `self.sequence_length_q25` entirely?

Reply:
replied in person. Not sure how much of details can be shared in public comment.